chainstate/src/config.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serde::{Deserialize, Serialize};
+
+/// The chainstate subsystem configuration.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ChainstateConfig {
+    /// The number of maximum attempts to process a block.
+    pub max_db_commit_attempts: usize,
```

Comment:
I'm not convinced commit is the right nomenclature here

Reply:
I know it's keeping the old name but still not a fan

Reply:
It was called slightly differently before: `MAX_DB_COMMIT_COUNT` and I thought that I was improved this name enough. 😅 Would `max_attempts_to_process_block` be better?

Reply:
I'm not sure whether you have an alternative word in mind, Ben, but "commit" is the word used in the context of ACID compliant databases. For example, see the option "synchronous_commit" in postgresql: https://postgresqlco.nf/doc/en/param/synchronous_commit/

Reply:
Discussed: this name will do for now.